### PR TITLE
Download plan query from resource URI

### PR DESCRIPTION
**Pull request by br648**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [x] Any modified or new methods or classes have helpful JavaDoc and code is thoroughly commented
- [x] The description lists all applicable issues this PR seeks to resolve
- [x] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing
- [x] The description lists all relevant PRs included in this release _(remove this if not merging to master)_

### Description

Update to download a plan query from https://raw.githubusercontent.com/ibi-group/open-source-configurations/main/atlanta/arc/planQuery.graphql.

### CODE REVIEW

File: src/main/java/org/opentripplanner/middleware/utils/GraphQLUtils.java [hunk 1]
```diff
@@ -1,52 +1,68 @@
...
     // Lazily-initialized in getPlanQueryTemplate()
     private static String planQueryTemplate = null;
 
     /**
-     * Location of the GraphQL plan query template file, as Java resource.
+     * Location of the GraphQL plan query template file, as URI resource.
      */
-    public static final String PLAN_QUERY_RESOURCE = "queries/planQuery.graphql";
+    private static final String PLAN_QUERY_RESOURCE_URI =
+        getConfigPropertyAsText("PLAN_QUERY_RESOURCE_URI", null);
```
Comment (binh-dam-ibigroup) [inline src/main/java/org/opentripplanner/middleware/utils/GraphQLUtils.java]: Should there be a default one? Either a default URL or default content?

**Review by JymDyerIBI — APPROVED** [review]
👍  Looks good to me.

File: src/main/java/org/opentripplanner/middleware/utils/GraphQLUtils.java (hunk 1 above)
Comment (br648) [reply src/main/java/org/opentripplanner/middleware/utils/GraphQLUtils.java]: So I have updated to include `https://raw.githubusercontent.com/ibi-group/open-source-configurations/main/atlanta/arc/planQuery.graphql` with the assumption any changes will be applied to this. It is worth bearing in mind that if another plan query is used, but is not defined, unexpected results will occur.

Comment (binh-dam-ibigroup) [reply src/main/java/org/opentripplanner/middleware/utils/GraphQLUtils.java]: No, that custom query template should not be the default.

Comment (br648) [reply src/main/java/org/opentripplanner/middleware/utils/GraphQLUtils.java]: @binh-dam-ibigroup do you have the default so I can upload and reference?

Comment (binh-dam-ibigroup) [reply src/main/java/org/opentripplanner/middleware/utils/GraphQLUtils.java]: @br648 Use the one that OTP-UI/OTP-RR use by default at https://raw.githubusercontent.com/opentripplanner/otp-ui/refs/heads/master/packages/core-utils/src/planQuery.graphql (we might move it in the future but for now it lives there)

Comment (br648) [reply src/main/java/org/opentripplanner/middleware/utils/GraphQLUtils.java]: Updated in: https://github.com/ibi-group/otp-middleware/pull/261/commits/695947a38032c6217feaa629bcd7c39ccb538485

File: configurations/default/env.yml.tmp [hunk 2]
```diff
@@ -93,4 +93,7 @@ TRIP_INSTRUCTION_UPCOMING_RADIUS: 10
 
 US_RIDE_GWINNETT_BUS_OPERATOR_NOTIFIER_API_URL: https://bus.notifier.example.com
 US_RIDE_GWINNETT_BUS_OPERATOR_NOTIFIER_API_KEY: your-key
-US_RIDE_GWINNETT_BUS_OPERATOR_NOTIFIER_QUALIFYING_ROUTES: agency_id:route_id
\ No newline at end of file
+US_RIDE_GWINNETT_BUS_OPERATOR_NOTIFIER_QUALIFYING_ROUTES: agency_id:route_id
+
+# The location for an OTP plan query request.
+PLAN_QUERY_RESOURCE_URI: https://plan.resource.com
```
Comment (binh-dam-ibigroup) [inline configurations/default/env.yml.tmp:108]: Please add this parameter to the config schema (and README).

File: src/main/java/org/opentripplanner/middleware/utils/GraphQLUtils.java [hunk 3]
```diff
@@ -1,52 +1,71 @@
...
 
     /**
-     * Location of the GraphQL plan query template file, as Java resource.
+     * Location of the GraphQL plan query template file, as URI resource.
      */
-    public static final String PLAN_QUERY_RESOURCE = "queries/planQuery.graphql";
+    private static final String PLAN_QUERY_RESOURCE_URI =
+        getConfigPropertyAsText(
+            "PLAN_QUERY_RESOURCE_URI",
+            "https://raw.githubusercontent.com/opentripplanner/otp-ui/refs/heads/master/packages/core-utils/src/planQuery.graphql"
```
Comment (binh-dam-ibigroup) [inline src/main/java/org/opentripplanner/middleware/utils/GraphQLUtils.java]: Extract a constant for this.

File: src/main/java/org/opentripplanner/middleware/utils/GraphQLUtils.java [hunk 4]
```diff
@@ -1,52 +1,71 @@
...
-                if (c == '\n') builder.append("\\n");
-                else if (c == '"') builder.append("\\\"");
-                else builder.append(c);
-            }
-        } catch (Exception e) {
-            LOG.error("Can't find \"{}\" resource.", resource, e);
+    static String getPlanQueryFromResource() {
+        if (GraphQLUtils.PLAN_QUERY_RESOURCE_URI == null) {
+            LOG.error("The plan query resource URI parameter \"PLAN_QUERY_RESOURCE_URI\" is undefined.");
+            return null;
```
Comment (binh-dam-ibigroup) [inline src/main/java/org/opentripplanner/middleware/utils/GraphQLUtils.java]: Does this apply anymore, now that there is a default URL?

File: src/main/java/org/opentripplanner/middleware/utils/GraphQLUtils.java [hunk 5]
```diff
@@ -1,52 +1,71 @@
...
 
     /**
-     * Return a GraphQL planQueryTemplate in Java string format, with {@code "} as {@code \"}
-     * @param resource the plan file or any GraphQL file
+     * Return a GraphQL planQueryTemplate in Java string format, with {@code "} as {@code \"}.
+     */
+    static String planQueryTemplateAsString() {
+        String rawPlanQuery = getPlanQueryFromResource();
+        if (rawPlanQuery == null) {
+            LOG.error("Unable to retrieve plan query from resource: {}.", GraphQLUtils.PLAN_QUERY_RESOURCE_URI);
```
Comment (binh-dam-ibigroup) [inline src/main/java/org/opentripplanner/middleware/utils/GraphQLUtils.java]: If this method returns null, how is that subsequently handled?

**Review by binh-dam-ibigroup — CHANGES_REQUESTED** [review]
Please add code to pass the query variable `mobilityProfile` to the GraphQL template.

File: src/main/java/org/opentripplanner/middleware/utils/GraphQLUtils.java (hunk 5 above)
Comment (br648) [reply src/main/java/org/opentripplanner/middleware/utils/GraphQLUtils.java]: Fixed in: https://github.com/ibi-group/otp-middleware/pull/261/commits/4539fc0123af8deee15cdd5ac87a6acc0c050a72

File: src/main/java/org/opentripplanner/middleware/utils/GraphQLUtils.java (hunk 4 above)
Comment (br648) [reply src/main/java/org/opentripplanner/middleware/utils/GraphQLUtils.java]: Fixed in: https://github.com/ibi-group/otp-middleware/pull/261/commits/4539fc0123af8deee15cdd5ac87a6acc0c050a72

File: src/main/java/org/opentripplanner/middleware/utils/GraphQLUtils.java (hunk 3 above)
Comment (br648) [reply src/main/java/org/opentripplanner/middleware/utils/GraphQLUtils.java]: Fixed in: https://github.com/ibi-group/otp-middleware/pull/261/commits/4539fc0123af8deee15cdd5ac87a6acc0c050a72

File: configurations/default/env.yml.tmp (hunk 2 above)
Comment (br648) [reply configurations/default/env.yml.tmp:108]: Fixed in: https://github.com/ibi-group/otp-middleware/pull/261/commits/4539fc0123af8deee15cdd5ac87a6acc0c050a72

**Comment by br648** [comment]
> Please add code to pass the query variable `mobilityProfile` to the GraphQL template.

@binh-dam-ibigroup I'm not entirely sure what you mean here? I've updated the template to include the `mobilityProfile` param:

https://github.com/ibi-group/open-source-configurations/blob/main/atlanta/arc/planQuery.graphql

I hope this is correct. Lemme know if you were expecting something else!

**Comment by binh-dam-ibigroup** [comment]
> > Please add code to pass the query variable `mobilityProfile` to the GraphQL template.
> 
> @binh-dam-ibigroup I'm not entirely sure what you mean here? I've updated the template to include the `mobilityProfile` param:
> 
> https://github.com/ibi-group/open-source-configurations/blob/main/atlanta/arc/planQuery.graphql
> 
> I hope this is correct. Lemme know if you were expecting something else!

During itinerary checks, the `mobilityProfile` field needs to be populated. As written, it is not (see screenshot):
image

**Comment by binh-dam-ibigroup** [comment]
> > > Please add code to pass the query variable `mobilityProfile` to the GraphQL template.
> > 
> > 
> > @binh-dam-ibigroup I'm not entirely sure what you mean here? I've updated the template to include the `mobilityProfile` param:
> > https://github.com/ibi-group/open-source-configurations/blob/main/atlanta/arc/planQuery.graphql
> > I hope this is correct. Lemme know if you were expecting something else!
> 
> During itinerary checks, the `mobilityProfile` field needs to be populated. As written, it is not (see screenshot)

To do that, I think you'll have to complete the implementation of `Otp2QueryParams.clone` to include `mobilityProfile`.

**Comment by br648** [comment]
> > > Please add code to pass the query variable `mobilityProfile` to the GraphQL template.
> > 
> > 
> > @binh-dam-ibigroup I'm not entirely sure what you mean here? I've updated the template to include the `mobilityProfile` param:
> > https://github.com/ibi-group/open-source-configurations/blob/main/atlanta/arc/planQuery.graphql
> > I hope this is correct. Lemme know if you were expecting something else!
> 
> During itinerary checks, the `mobilityProfile` field needs to be populated. As written, it is not (see screenshot): image

Fixed in: https://github.com/ibi-group/otp-middleware/pull/261/commits/e203492294f2fc48eeb66197c292d0ff97eb40c7

**Review by binh-dam-ibigroup — APPROVED** [review]
Nice, it works with custom plan query and the default one.

**Pull request merged**